Jupyter Notebook Previewer

JR-1991 opened the pull request:
Jupyter notebooks are widely used in our community, but there hasn’t been a previewer for them yet, aside from a full Binder run. This pull request introduces a Jupyter Notebook previewer to the Dataverse external tools ecosystem, allowing users to view `.ipynb` files directly in the browser with syntax highlighting, markdown support, and mathematical expression rendering. In addition to typical Python, this previewer also utilizes Prism to provide syntax highlighting for Julia and R.

**Jupyter Notebook Previewer Integration**

* Added a new external tool registration example for the Jupyter Notebook previewer in both `5.2curlcommands.md` and `6.1curlcommands.md` documentation files, detailing its usage and API parameters. [[1]](diffhunk://#diff-2b81dad3a1adc803d93f4d62422839b67be73cb0e69e9d85d011a65f749ad93eR747-R771) [[2]](diffhunk://#diff-574e60aa6157a13d4423316f4460636bb2a1a11ea628936c689741f683ce9b84R1314-R1339)

**Frontend Implementation**

* Introduced `previewers/betatest/JupyterPreview.html`, a standalone HTML page that loads required libraries (Prism.js, marked.js, notebookjs, KaTeX, Bootstrap) and custom scripts/styles to render Jupyter notebooks with syntax highlighting, markdown, and math support.
* Added `previewers/betatest/js/jupyter.js`, which configures notebook rendering, syntax highlighting, and handles the loading and display of notebook content.
* Created `previewers/betatest/css/jupyter.css` for custom styling of notebook cells, headings, tables, and outputs to mimic the Jupyter Notebook look and feel.

**Internationalization**

* Updated English, Spanish, and French translation files to include the `jupyterPreviewText` key for multi-language support of the previewer interface. [[1]](diffhunk://#diff-f5e7135c6049302afeb65b344d1bf98b499bead740c760fa8cdc752ed219a586R25) [[2]](diffhunk://#diff-fca82b35b084edd34c7f9cf730769de8c4308748cb525683a335a7e2bbb1a1d8R24) [[3]](diffhunk://#diff-c74ca3afee3c9ac07a6a17834dada12203bc19e66ecbdbab8499d8cd262ddc56R25)

**Local tests**

I’ve tested this previewer on a locally hosted Dataverse 6.6 container. I’ve been testing older releases in parallel, so the older release is being used here. However, I don’t think this will affect newer versions. I have used the following manifest:

```json
{
    "displayName":"Show Jupyter Notebook",
    "description":"View the Jupyter Notebook file.",
    "toolName":"jupyterPreviewer",
    "scope":"file",
    "types":["preview"],
    "toolUrl":"http://localhost:8000/JupyterPreview.html",
    "toolParameters": {
        "queryParameters":[
          {"fileid":"{fileId}"},
          {"siteUrl":"{siteUrl}"},
          {"datasetid":"{datasetId}"},
          {"datasetversion":"{datasetVersion}"},
          {"locale":"{localeCode}"},
          {"key":"{apiToken}"}
        ]
      },
    "contentType":"application/x-ipynb+json"
  }
```

**Screenshots**

## Draft View

<img width="1728" height="1080" alt="draftView" src="https://github.com/user-attachments/assets/6bd9e614-4f70-4a44-bf39-d3489216804b" />

## Published View

<img width="1728" height="1080" alt="publishView" src="https://github.com/user-attachments/assets/ae24b2b0-5d1d-4b09-b0b5-75f4b15a71ff" />

## Full View

<img width="1728" height="1080" alt="fullView" src="https://github.com/user-attachments/assets/9c5e38ef-0f9d-4308-9753-ef0fe59553ea" />

## Code and Viz Example

<img width="1172" height="726" alt="image" src="https://github.com/user-attachments/assets/ef7b6042-e87f-4abb-a9ab-c57c273b522f" />

JR-1991 committed:
Add Jupyter Notebook previewer

Introduces a new previewer for Jupyter Notebook (.ipynb) files, including HTML, CSS, and JavaScript for rendering notebooks with syntax highlighting, markdown, and math support. Enhances user experience by mimicking native Jupyter styling and supporting multiple programming languages.

JR-1991 committed:
Add Jupyter Notebook previewer curl command

Added example curl commands for registering a Jupyter Notebook previewer external tool in both 5.2curlcommands.md and 6.1curlcommands.md. This enables users to preview .ipynb files via an external tool integration.

JR-1991 committed:
Add Jupyter Notebook Preview translation key

Introduces the 'jupyterPreviewText' key to English, Spanish, and French i18n files to support Jupyter Notebook preview localization.

JR-1991 committed:
Fix CSS comment formatting in jupyter.css

Updated multi-line comments in jupyter.css to use correct CSS comment syntax, improving readability and preventing potential parsing issues.

qqmyers commented:
@JR-1991 - Cool!. Not sure I understand what you're asking about signed urls: When using retriever.js, and using writeContentAndData() in your js, both forms are automatically supported. The only change required is to change the configuration json. In general, I think we've been putting the old form in the 5.2 curl commands (prior to signed urls) and using the signed url form in the 6.1+ examples. 

For signed URLs, using retriever.js, you have to add URLs retriever.js expects - as in https://github.com/gdcc/dataverse-previewers/blob/a7472ad8f433d40a2809d0ced2db9baf629e9942/6.1curlcommands.md?plain=1#L25-L44 (you can see the names for these URLs are hardcoded at https://github.com/gdcc/dataverse-previewers/blob/a7472ad8f433d40a2809d0ced2db9baf629e9942/previewers/betatest/js/retriever.js#L43-L51). (I don't think the tool parameters are used at all with signed Urls, but we generally haven't removed them from the example curl calls.)

JR-1991 commented:
@qqmyers, thanks for clarifying and providing the snippets. I was only looking at the lower part implementing the previous mechanism, but that’s fantastic to know! I’ll add the `allowedApiCalls` and open it for review.

JR-1991 committed:
Add allowedApiCalls to JupyterPreview tool config

Extended the JupyterPreview tool configuration with an allowedApiCalls array specifying permitted API calls for file retrieval and dataset metadata. Removed toolParameters and updated formatting for clarity.

qqmyers committed:
Merge branch 'develop' into jupyter-notebook-preview

pdurbin commented:
Added to https://docs.google.com/document/d/10RIvsUntk9vbCofpic-ld-yrfko3sUvH8uH4eqqfYco/edit?usp=sharing
